src/Impute.jl

```diff
@@ -1,14 +1,40 @@
 module Impute
 
-using DataFrames
+using IterTools
 using Statistics
+using StatsBase
+using Tables: Tables, materializer, istable
 
-import DataFrames: DataFrameRow
-import Base.Iterators
+import Base.Iterators: drop
 
 export impute, impute!, chain, chain!, drop, drop!, interp, interp!, ImputeError
 
-const Dataset = Union{AbstractArray, DataFrame}
+function __init__()
+    sym = join(["chain", "chain!", "drop", "drop!", "interp", "interp!"], ", ", " and ")
+
+    @warn(
+        """
+        The following symbols will not be exported in future releases: $sym.
+        Please qualify your calls with `Impute.<method>(...)` or explicitly import the symbol.
+        """
+    )
+
+    @warn(
+        """
+        The default limit for all impute functions will be 1.0 going forward.
+        If you depend on a specific threshold please pass in an appropriate `AbstractContext`.
+        """
+    )
+
+    @warn(
+        """
+        All matrix imputation methods will be switching to the column-major convention
```

Comment:
Are you sure you want to move to that convention in the future? It goes against `cor` in Statistics and the fact that Tables.jl objects store observations in rows. Also in terms of performance, I'd guess that since arrays are column-major it's faster to store variables in columns when you operate on one variable at a time.

Reply:
My main reason for wanting to switch this is that many of the multivariate imputation methods I'm looking at implementing would use a multivariate distribution from Distributions.jl or MultivariateStats.jl which follow that convention. I suppose those methods could convert the data as necessary... I also generally prefer that convention because you already get the alternate convention with a table structure.

Reply:
I've reverted this since enough people seem to have concerns about it and I don't have implemented methods that would benefit from the change yet.